File: parentheses.cpp
#include<iostream>
#include<string.h>
#include<stack>
using namespace std;
bool balance(string s)
{	
int length=s.length();
	cout<<"The lenght is "<<length<<endl;
	stack<char> mystack;
	char a;
	for(int i =0;i<length;i++)
	{
		a=s[i];
		if((a=='(')||(a=='{')||(a=='['))
		{
			mystack.push(a);
		}
		if((a==')'))
		{
			if(mystack.top()!='(') 
			{
				return false;
			}
			mystack.pop();
				
		}
		if((a==']'))
		{
			if(mystack.top()!='[') 
			{
				return false;
			}
			mystack.pop();
				
		}
		if((a=='}'))
		{
			if(mystack.top()!='{') 
			{
				return false;
			}
			mystack.pop();	
		}
		
	}
	if(!mystack.empty())
		return false;
	return true;
}
int main()
{
string expression="[()]";
cout<<"Expression is "<<expression<<endl;
	cout<<balance(expression)<<endl;;
	return 0;
}

File: Parentheses.java
import java.util.*;
class Parentheses
{

public boolean isBalanced(String expression)
{
	
	Stack<Character> stck=new Stack<Character>();
	char s[]=expression.toCharArray();
	char a;
	Integer length=expression.length();
	for(int i=0;i<length;i++)
	{
		a=s[i];
		if((a=='(')||(a=='{')||(a=='['))
		{
			stck.push(a);
		}
		if((a==')'))
		{
			if(stck.pop()!='(') 
			{
				return false;
			}	
		}
		if((a==']'))
		{
			if(stck.pop()!='[') 
			{
				return false;
			}
		}
		if((a=='}'))
		{
			if(stck.pop()!='{') 
			{
				return false;
			}
		}
		
	}
	if(!stck.isEmpty())
		{
			return false;
		}
		return true;
}
public static void main(String args[])
{
Parentheses p =new Parentheses();
String expression="[]";
System.out.println(p.isBalanced(expression));
System.exit(0);
}

}